### PR TITLE
Update MLIR hash to latest MLIR

### PR DESCRIPTION
**Pull request by math-fehr**

### CODE REVIEW

**Comment by webmiche** [comment]
image

**Comment by webmiche** [comment]
[RFC](https://discourse.llvm.org/t/rfc-introduce-a-new-dense-array-attribute/63279/8), [commit](https://github.com/llvm/llvm-project/commit/508eb41d82ca)

**Review by superlopuh — APPROVED** [review]
🚢

File: xdsl/parser.py
Comment (webmiche) [inline xdsl/parser.py:729]: I dont think you need the `is not None` part, right?

File: xdsl/parser.py
Comment (webmiche) [inline xdsl/parser.py:945]: Maybe check that the parsed attribute makes sense? (Is integer or float type)

File: xdsl/parser.py
Comment (webmiche) [inline xdsl/parser.py:1037]: Dito.

**Review by webmiche — APPROVED** [review]
LGTM overall!

File: xdsl/parser.py
Comment (superlopuh) [reply xdsl/parser.py:729]: `dense` might not be None but be another falsey value, the `is not None` is better. The type checker prefers it also in most cases

Comment (math-fehr) [reply xdsl/parser.py:729]: So this is something that I know @georgebisbas has different opinion than me.
I think that `is not None` is more explicit, and way less error prone, since we won't break it because of a refactor for instance. Though yes, it is redundant.

Since we already uniformized the code by removing the `is not None` in the other files, I'll remove it here as well to be coherent!

Comment (math-fehr) [reply xdsl/parser.py:729]: I would argue that we should always have `not None`, since this is more robust in general, and convey the right information.
We want to check that something is not `None`, not that something is null, if that makes sense. Even if `is null` == `is None` in these cases, if we change our API this will break without any warning/errors.

**Comment by webmiche** [comment]
I worked on the tests a bit. Weirdly enough, there seems to have been a cache miss on one of my recent pushes. Not sure why, looking into it...

**Comment by webmiche** [comment]
Currently, there is an issue with initialized global memrefs. As the initial_value they expect a DenseArrayBase or a UnitAttr, but if I give it a DenseArrayBase, it wants an elements attribute.

I changed he initial_value to a unit attr for the moment...

File: xdsl/dialects/memref.py
Comment (superlopuh) [inline xdsl/dialects/memref.py:221]: Is this PR still WIP?

Comment (webmiche) [reply xdsl/dialects/memref.py:221]: There is only this issue left, which relates to the issue I mentioned above.

**Comment by math-fehr** [comment]
Thanks! I'll take a look at it!

**Comment by math-fehr** [comment]
Note, I would like to merge this only after the new parser is merged.
Otherwise, we'll just end up with more changes in the parser patch, which will delay it again!

**Comment by math-fehr** [comment]
(Converting to draft while I'm rebasing everything, and fixing bugs)

**Comment by math-fehr** [comment]
I rebased it for after the parser was merged, and I fixed the `global` issue (btw, it should have been `dense`, not `array` that was expected).
Is the fix good for you @webmiche? If that's the case I'll merge it

File: xdsl/dialects/memref.py
Comment (webmiche) [inline xdsl/dialects/memref.py:245]:
```suggestion
                "Global initial value is expected to be a dense type or a UnitAttr.")
```

File: xdsl/dialects/builtin.py
Comment (webmiche) [inline xdsl/dialects/builtin.py:643]: I guess this could use some verifiers, do the types even work and so on.

**Review by webmiche — APPROVED** [review]
Nice, LGTM. Removing the mlir converter already makes us less dependent on the bindings. We are btw still using them to check whether we execute certain tests or not, but I guess we can change that in a different PR in the future.

Also, the name `with-bindings` for one of the test folders doesn't make much sense anymore, though it still uses the bindings to check whether to execute or not, so I guess that's fine :sweat_smile:

**Comment by math-fehr** [comment]
Should be good now!
I renamed `with-bindings` to `with-mlir`, but I'm not quite sure how we can check the existence of `mlir` without going through the bindings

**Comment by math-fehr** [comment]
We could also look at it in another PR ;)

**Comment by webmiche** [comment]
Yeah, I agree with that. I guess we can check for mlir-opt being in the path. This is pretty much the remaining condition to run the tests.

Let's move this to a different PR.

**Review by webmiche — APPROVED** [review]

**Pull request merged**
